docs/src/spaces.md

```diff
@@ -0,0 +1,155 @@
+
+## Frequently-Accessed Caching Questions
+
+> Can I trigger data regeneration if the space is found to be empty/files are missing?
+
+Yes, this is quite simple; just check the contents of the directory when you first call `get_space!()`, and if it's empty, run your generation function:
+
+```julia
+using Pkg, Pkg.Spaces
+
+function get_dataset_dir()
+    dataset_dir = @get_space!("dataset")
+    if isempty(readdir(dataset_dir))
+        perform_expensive_dataset_generation(dataset_dir)
+    end
+    return dataset_dir
+end
+```
+
+> Can I create a scratchspace that is not shared across versions of my package?
+
+Yes!  Make use of the `key` parameter and Pkg's ability to look up the current version of your package at compile-time:
+
+```julia
+module VersionSpecificExample
+using Pkg, Pkg.Spaces
+
+# Helpers to get current package UUID and VersionNumber
+function get_uuid()
+    return Base.PkgId(@__MODULE__).uuid
+end
+function get_version(uuid)
+    ctx = Pkg.Types.Context()
+    uuid, entry = first(filter(((u, e),) -> u == uuid, ctx.env.manifest))
+    return entry.version
+end
+
+# Get the current version at compile-time, that's fine it's not going to change. ;)
+const v = get_version(get_uuid())
+
+# This will be filled in by `__init__()`; it might change if we get deployed somewhere
+version_specific_space = ""
+
+function __init__()
+    # This space will be unique between versions of my package that different major and
+    # minor versions, but allows patch releases to share the same.
+    global version_specific_space = @get_space!("data_for_version-$(v.major).$(v.minor)")
+end
+
+end # module
+```
+
+> Can I use a scratchspace as a temporary workspace, then turn it into an Artifact?
+
+Yes!  Once you're satisfied with your dataset that has been cooking inside a space, and you're ready to share it with the world as an immutable artifact, you can use `create_artifact()` to create an artifact from the space, `archive_artifact()` to get a tarball that you can upload somewhere, and `bind_artifact!()` to write out an `Artifacts.toml` that allows others to download and use it:
+
+```julia
+using Pkg, Pkg.Spaces, Pkg.Artifacts
+function export_space(space_name::String)
+    space_dir = @get_space!(space_name)
+
+    # Copy space directory over to an Artifact
+    hash = create_artifact() do artifact_dir
+        rm(artifact_dir)
+        cp(space_dir, artifact_dir)
+    end
+
+    # Archive artifact out to a tarball
+    mktempdir() do upload_dir
+        tarball_path = joinpath(upload_dir, "$(space_name).tar.gz")
+        tarball_hash = archive_artifact(hash, tarball_path)
+
+        # Upload tarball to a hosted site somewhere.  Note; this function does not exist:
```

Comment:
Perhaps this should be written in a way that would make it easy to manually upload the tarball as well. Now with the `mktempdir` it would delete the directory when the scope exit so there is no point where you can manually do things.